src/mqtt/time.rs

```diff
@@ -1,6 +1,8 @@
 use chrono::NaiveDateTime;
+use serde::Serialize;
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(untagged)]
 pub enum Time {
     Retained,
```

Comment:
Currently, the untagged output of Retained is `null`… It's not ideal but at the same time its somewhat clear what it means, so I guess we can stick with it?

Reply:
Not including it might be even more confusing… Guess null is both easy to understand and simple with the code.